boards/arc/qemu/qemu.dtsi

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2020, Synopsys, Inc. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+
+#include "skeleton.dtsi"
+
+/ {
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		intc: arcv2-intc {
+			compatible = "snps,arcv2-intc";
+			interrupt-controller;
+			#interrupt-cells = <2>;
+		};
+	};
+
+	flash0: flash@80000000 {
+		compatible = "soc-nv-flash";
+		reg = <0x80000000 (256*1024)>;
+	};
+
+	sram0: sram@80040000 {
+		device_type = "memory";
+		compatible = "mmio-sram";
+		reg = <0x80040000 (256*1024)>;
+	};
+
+	ns16550_uart: uart@f0000000 {
+		compatible = "ns16550";
+		clock-frequency = <10000000>;
+		reg = <0xf0000000 0x400>;
+		current-speed = <115200>;
+		label = "UART_0";
+		interrupt-parent = <&intc>;
+		interrupts = <24 1>;
+	};
+
+	chosen {
```

Comment:
My idea was to keep as much as possible in the common .dtsi because what we have here is basically a virtual board with swappable CPU: we either use ARC EM-based CPU or ARC HS-based one. Note EM & HS are binary incompatible even though share the same (or very similar) ISA and so I don't have 2 completely separate boards for them. And that's what we have in case of nSIM as well BTW.